core/arch/x86_64/kernel/thread.c

```diff
@@ -732,7 +757,8 @@ void thread_state_save(vaddr_t sp)
 	assert(threads[ct].ta_idx < MAX_TA_IDX+1);
 	threads[ct].stack_va_curr[threads[ct].ta_idx-1] = sp;
 
-	syscall_init(sp - 64);
+	syscall_init(sp - 64, threads[ct].svc_table);
```

Comment:
why we need set MSR in thread_state_save()? in save, we should "save" something, not "set"